src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -53,16 +53,22 @@ public Task<JObject> GetAppServiceEnvironmentDetails(string hostingEnvironmentNa
 
         public async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string siteName)
         {
-            var response = await GetObserverResource($"sites/{siteName}/runtimesiteslotmap");
-            var slotTimeRangeCaseSensitiveDictionary = JsonConvert.DeserializeObject<Dictionary<string, List<RuntimeSitenameTimeRange>>>(response);
-            var slotTimeRange = new Dictionary<string, List<RuntimeSitenameTimeRange>>(slotTimeRangeCaseSensitiveDictionary, StringComparer.CurrentCultureIgnoreCase);
-            return slotTimeRange;
+            return await GetRuntimeSiteSlotMap(null, siteName);
         }
 
         public async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string stampName, string siteName)
         {
-            var response = await GetObserverResource($"stamp/{stampName}/sites/{siteName}/runtimesiteslotmap");
-            var slotTimeRangeCaseSensitiveDictionary = JsonConvert.DeserializeObject<Dictionary<string, List<RuntimeSitenameTimeRange>>>(response);
+            return await GetRuntimeSiteSlotMapInternal(stampName, siteName);
+        }
+
+        private async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMapInternal(string stampName, string siteName)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, string.IsNullOrEmpty(stampName) ? $"/sites/{siteName}/runtimesiteslotmap" : $"stamp/{stampName}/sites/{siteName}/runtimesiteslotmap");
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", await GetAccessToken(_configuration.RuntimeSiteSlotMapResourceUri));
+            var response = await _httpClient.SendAsync(request);
+            response.EnsureSuccessStatusCode();
```

Comment:
Do we want to handle errors in a way that we can log any error coming from the API?